Code/Tools/AssetProcessor/native/ui/MainWindow.cpp

```diff
@@ -864,6 +876,79 @@ void MainWindow::SaveLogPanelState()
     }
 }
 
+void MainWindow::ShowOutgoingProductDependenciesContextMenu(const QPoint& pos)
+{
+    using namespace AssetProcessor;
+    auto productAt = [this](const QPoint& pos)
+    {
+        const QModelIndex assetIndex = ui->productAssetDetailsPanel->GetOutgoingProductDependenciesTreeView()->indexAt(pos);
```

Comment:
is it possible for GetOutgoingProductDependenciesTreeView()->indexAt() to return an invalid index?

Reply:
This function is invoked only when `QWidget::customContextMenuRequested` signal is emitted by `OutgoingProductDependenciesTreeView` (see [MainWindow.cpp:432](https://github.com/aws-lumberyard-dev/o3de/blob/sven_ProductAssetDetail/Code/Tools/AssetProcessor/native/ui/MainWindow.cpp#L432)), and the argument, `pos`, is the position where the request (right click, I guess) happened. Since the `pos` is filled in by the signal, it is normally inside the widget clicked, and the `indexAt(pos)` should always return a valid index. 

[Qt's doc](https://doc.qt.io/qt-5/qwidget.html#customContextMenuRequested) said though, that if the widget is a [QAbstractScrollArea](https://doc.qt.io/qt-5/qabstractscrollarea.html) "that map the context menu event to coordinates of the [viewport()](https://doc.qt.io/qt-5/qabstractscrollarea.html#viewport)", `pos` might not fall inside the widget. I don't think that is the case here.
@AMZN-stankowi , I ported this code from your branch. Can you help me confirming if this is correct?

Reply:
This applies to `GetIncomingProductDependenciesTreeView()->indexAt()` as well.

Reply:
I think this could be updated to verify that indexAt returns a valid index. You can call assetIndex.isValid() before attempting to cast the internal pointer to a ProductDependencyTreeItem.

I can't think of a case where the context menu would pop up without having a valid position, but it's better to handle that case cleanly instead of what would happen now, the application would probably crash.

Reply:
finished